### DISCUSSION TITLE
Can the corner-point grid be converted to a format usable in libmesh?

### DISCUSSION THREAD

**Discussion by tu1620808124** [libMesh/libmesh#3646] [Q&A] [answered]
Hello,
I'd like to perform reservoir simulation using libmesh, but most of the available data sets are in corner-point grid format. Is there a method to convert these corner-point grids into a format compatible with libmesh?

**Comment by jwpeterson** [comment, marked as answer]
Based on the [Wikipedia article](https://en.wikipedia.org/wiki/Corner-point_grid) it sounds like a type of structured hexahedral mesh, so, yes I think you could write some code that would import such a mesh relatively easily. On the libMesh side, the mesh would of course be treated as unstructured HEX8, HEX20, or HEX27 elements. It would be simplest if the grid you want to read is in an ASCII format, and not a binary blob. You can look at one of the other ASCII-based mesh readers like [AbaqusIO](https://github.com/libMesh/libmesh/blob/52f92e583fdb6d91e46a2b85aa36684275e51849/src/mesh/abaqus_io.C) to get some ideas on how to read from a file and add Nodes/Elems to the Mesh.
